arangod/IResearch/IResearchView.cpp

```diff
@@ -316,6 +316,117 @@ inline void insertDocument(
   doc.insert(irs::action::store, primaryKey);
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// @brief syncs an IResearch DataStore if required
+/// @return a sync was executed
+////////////////////////////////////////////////////////////////////////////////
+bool syncStore(
+    irs::directory& directory,
+    irs::directory_reader& reader,
+    irs::index_writer& writer,
+    std::atomic<size_t>& segmentCount,
+    arangodb::iresearch::IResearchViewMeta::CommitMeta::ConsolidationPolicies const& policies,
+    bool forceCommit,
+    bool runCleanupAfterCommit,
+    std::string const& viewName
+) {
+  char runId = 0; // value not used
+
+  // ...........................................................................
+  // apply consolidation policies
+  // ...........................................................................
+
+  for (auto& entry: policies) {
+    if (!entry.segmentThreshold()
+        || entry.segmentThreshold() > segmentCount.load()) {
+      continue; // skip if interval not reached or no valid policy to execute
+    }
+
+    LOG_TOPIC(DEBUG, arangodb::iresearch::IResearchFeature::IRESEARCH)
+      << "registering consolidation policy '" << size_t(entry.type()) << "' with IResearch view '" << viewName << "' run id '" << size_t(&runId) << " segment threshold '" << entry.segmentThreshold() << "' segment count '" << segmentCount.load() << "'";
+
+    try {
+      writer.consolidate(entry.policy(), false);
+      forceCommit = true; // a consolidation policy was found requiring commit
+    } catch (std::exception const& e) {
+      LOG_TOPIC(WARN, arangodb::iresearch::IResearchFeature::IRESEARCH)
+        << "caught exception during registeration of consolidation policy '" << size_t(entry.type()) << "' with IResearch view '" << viewName << "': " << e.what();
+      IR_EXCEPTION();
```

Comment:
`IR_EXCEPTION` is a deceptive name. I assumed this threw another exception, but it seems to just log something. Maybe consider renaming it in the future? `IR_LOG_EXCEPTION()`?

Reply:
will fix it in iresearch